python/ml4ir/base/tests/test_ml4ir_sanity.py

```diff
@@ -203,14 +188,8 @@ def test_linear_ml4ir_sanity_7(self):
                         log_regression_mrr=0.4938108725899423,
                         working_dir=pathlib.Path(self.working_dir.path), log_dir=pathlib.Path(self.log_dir))
 
-    def _test_linear_ml4ir_sanity_8(self):
+    def test_linear_ml4ir_sanity_8(self):
```

Comment:
Some of these tests were deactivated earlier due to memory issues. I am reactivating them now.

Reply:
These tests may run for up to 150 epochs with batch size of 1. This is in line 46-48. 
How about doing 2-3 epochs with larger batches? Can be a followup WI. 
We should be also reconsidering if all these tests (with probably copy-paste numbers) indeed add value. 
Thoughts? 
cc: @mohazahran since you have added them. If you have a fresher look on this, please chime in

Reply:
Yep, would like to get @mohazahran 's thoughts on updating these.

I'm for deprecating a few of these.

Reply:
Initially this PR was training all the models (linear regression, perceptron and ml4ir0 on the spot, so it was easier to configure the comparison between them by changing the parameters (batch size, training set size ...). but since LR and perceptron required extra dependencies only their weights are used after training them offline. So updating these will required re-training these models and updating the weights.
One possible this we can do is we can retire the perceptron (as it's inferior to the other two models) but i dont think this will really help the run time.